desktop-exporter/app/utils/array-to-tree.ts

```diff
@@ -31,21 +35,24 @@ export function arrayToTree(spans: SpanData[]): TreeItem[] {
       };
     }
 
+    let treeItem = lookup[spanID];
+
     // Note A:
     // If the span has been added to the lookup structure as a missing/incomplete parent
     // on a previous pass (see Note B), update it and mark it present, and remove it from the missing set.
-    if (lookup[spanID].status === SpanDataStatus.missing) {
-      let children = lookup[spanID].children;
-      lookup[spanID] = {
+    if (treeItem.status === SpanDataStatus.missing) {
```

Comment:
Instead of always referring to `lookup[spanID]`, if we reference `treeItem` here then the type system can follow that we are making assertions against it and track the flow